Source/Bugsnag/BugsnagNotifier.m

```diff
@@ -51,13 +64,44 @@
 
 static struct bugsnag_data_t g_bugsnag_data;
 
-void serialize_bugsnag_data(const KSCrashReportWriter *writer) {
+/**
+ *  Writes a dictionary to a destination using the KSCrash JSON encoding
+ *
+ *  @param dictionary  data to encode
+ *  @param destination target location of the data
+ */
+void BSSerializeJSONDictionary(NSDictionary *dictionary, char **destination) {
+    NSError *error;
+    NSData *json = [KSJSONCodec encode: dictionary options:0 error:&error];
+
+    if (!json) {
+        NSLog(@"Bugsnag could not serialize metaData: %@", error);
+        return;
+    }
+
+    *destination = reallocf(*destination, [json length] + 1);
+    if (*destination) {
+        memcpy(*destination, [json bytes], [json length]);
+        (*destination)[[json length]] = '\0';
+    }
+}
+
+/**
+ *  Handler executed when the application crashes. Writes information about the
+ *  current application state using the crash report writer.
+ *
+ *  @param writer report writer which will receive updated metadata
+ */
+void BSSerializeCrashData(const KSCrashReportWriter *writer) {
     if (g_bugsnag_data.configJSON) {
         writer->addJSONElement(writer, "config", g_bugsnag_data.configJSON);
     }
     if (g_bugsnag_data.metaDataJSON) {
         writer->addJSONElement(writer, "metaData", g_bugsnag_data.metaDataJSON);
     }
+    BugsnagNotifier* notifier = [Bugsnag notifier];
+    [notifier addCrashDataToStateWithSeverity:BugsnagSeverityError andDepth:4];
+    [notifier metaDataChanged:notifier.state];
```

Comment:
It wasn't the state modification which was important, just [writing the `state` JSON](https://github.com/bugsnag/bugsnag-cocoa/blob/kattrali/send-severity-breadcrumbs-with-uncaught/Source/Bugsnag/BugsnagNotifier.m#L206). It could be called directly/referenced though.